### DISCUSSION TITLE
error when opening some ang files

### DISCUSSION THREAD

**Discussion by cegriesbach** [mtex-toolbox/mtex#1177] [Ask Anything]
Hello all,

I am unable to open select .ang files using the loadEBSD_ang command. The following error message is displayed: "EBSD format 'ANG' does not match the data." 

I have attached two example .ang files: the "20210708_AgSet20_S1_Area 2_OIM Map 1rescan.ang" file opens without error, while the error occurs for the "20210817_Set26S2_3_Rescan1.ang" file (this error also occurs with several other files as well). There are two differences I noticed in the files that cause the error: i) different phases (user defined in ATEX), ii) 14 data columns instead of 13.

I have saved the ang files after reindexing in OIM V8. I am using Matlab version R2020b and MTEX version 5.7. I have looked at several other discussions in this forum and the old one regarding issues with loading ang files, but it appears all the bug fixes for those are incorporated into the MTEX 5.7 version of loadEBSD_ang.m.

[example ang files.zip](https://github.com/mtex-toolbox/mtex/files/7033720/example.ang.files.zip)

Any help would be appreciated. Thanks!

-Claire

**Comment by lfgmorales** [comment]
Hi Claire

The easiest (but not elegant) would be delete the additional column, did you run EDS in this specific map?

Cheers

Luiz

Luiz F. G. Morales
ScopeM - ETH Zürich
Sent from iPhone

On 23 Aug 2021, at 19:11, cegriesbach ***@***.***> wrote:

﻿

Hello all,

I am unable to open select .ang files using the loadEBSD_ang command. The following error message is displayed: "EBSD format 'ANG' does not match the data."

I have attached two example .ang files: the "20210708_AgSet20_S1_Area 2_OIM Map 1rescan.ang" file opens without error, while the error occurs for the "20210817_Set26S2_3_Rescan1.ang" file (this error also occurs with several other files as well). There are two differences I noticed in the files that cause the error: i) different phases (user defined in ATEX), ii) 14 data columns instead of 13.

I have saved the ang files after reindexing in OIM V8. I am using Matlab version R2020b and MTEX version 5.7. I have looked at several other discussions in this forum and the old one regarding issues with loading ang files, but it appears all the bug fixes for those are incorporated into the MTEX 5.7 version of loadEBSD_ang.m.

example ang files.zip<https://github.com/mtex-toolbox/mtex/files/7033720/example.ang.files.zip>

Any help would be appreciated. Thanks!

-Claire

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/1177>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN22EWFZCLD7T2TWINTT6J6MXANCNFSM5CVALVAA>.

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/1177", "url": "https://github.com/mtex-toolbox/mtex/discussions/1177", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Reply by cegriesbach** [reply]
Hi Luiz,

Thanks for the suggestion. I deleted the 14th column (and edited the header to only specify 13 columns) but the error persists. I did not run eds on this map, so I am not sure where the 14th column is coming from. It is labeled as "Custom Value." However, it appears the error is originating from something else. (I have added the edited ang file to the attached folder.)

Any other suggestions?

Thanks,
Claire

[example ang files_new.zip](https://github.com/mtex-toolbox/mtex/files/7041626/example.ang.files_new.zip)

**Comment by lfgmorales** [comment]
Hi Claire


Although I can open your data on OIM, I cannot get rid of the last column, not even by re-exporting. Is this dataset collect on TEM Astar Nanomegas? The heads of the files are also totally different, the one that works has a reasonable number of reflectors, while the one that is messed up has like 2000 lines of reflector data, although this should interfere on the exporting to *ang files, something you may want to look.



cheers



L





Luiz F. G. Morales
ScopeM & Structural Geology and Tectonics Group
ETH Zürich
________________________________

**Comment by lfgmorales** [comment]
Hi again


Your problematic file has some issues in the header (somewhere between the start and the beginning of the data), I cannot pinpoint where exactly, but if I delete the last column and substitute the header of your problematic map with the header of the file name "20210708_AgSet20_S1_Area 2_OIM Map 1rescan" it works (of course, now considering the phases FCC, BCC and silver that you have on the later. The file is attached


cheers



L





Luiz F. G. Morales
ScopeM & Structural Geology and Tectonics Group
ETH Zürich
________________________________

**Reply by cegriesbach** [reply]
Hi Luiz,

Thanks for your help with troubleshooting. I am using EDAX's APEX software to collect the TKD maps. The software auto generates the list of reflectors based on the material .cif files I provided, since I wanted to include rare 2H and 4H polytypes of silver that could be present in my sample. I do not know why the list of reflectors is so long, but I think this is what is causing issues. I did a bit of debugging and found that in the loadEBSD_ang function, a maximum of 2000 lines is read in for the header. Additionally, the function txt2mat is routinely used in the loadEBSD_ang, loadEBSD_generic, and load_generic functions: this function tries to find the number of header lines if not given, and I think the long list of reflectors in the header causes an issue with this. Sometimes a maximum number of lines of 10000 is also specified in the txt2mat function option 'RowRange', which this data set exceeds. If I delete most of the reflector lines in the ang file, it loads without error. 

I think I will try to edit my phases used for indexing so as to avoid this issue. But I thought I would point out the issue here in case anyone has a similar problem with large data sets.

Thanks for your help!

-Claire
